### PR TITLE
Upgrade to log-collection v0.28.0

### PR DESCRIPTION
**Pull request by djaglowski**
Updates log-collection library to v0.28.0.

### CODE REVIEW

**Review by pmm-sumo — APPROVED** [review]

**Comment by jpkrohling** [comment]
Waiting on today's release before merging this. It also means that this PR will get outdated (changelog, go.mod/sum).

File: CHANGELOG.md
Comment (pmm-sumo) [reply CHANGELOG.md:17]: Yeah, I was just bit unsure but it works for me :)

**Comment by jpkrohling** [comment]
PR updated with the changes post-v0.48.0

**Review by codeboten — APPROVED** [review]

**Comment by politician** [comment]
@djaglowski Do not forget to update the examples https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/examples/kubernetes

**Pull request merged**
